tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -155,6 +155,32 @@ bool TypeTranslator::is1xNMatrixType(QualType type) {
   return rowCount == 1 && colCount > 1;
 }
 
+bool TypeTranslator::isScalarMatOrVectorMat(QualType type) {
+  return is1x1MatrixType(type) || isSingleRowOrSingleColMatrix(type);
+}
+
+bool TypeTranslator::isSingleRowOrSingleColMatrix(QualType type,
```

Comment:
It would be nice to rename these methods to have consistent names like: `is1x1Matrix`, `isMx1Matrix`, `is1xNMatrix`, `isMxNMatrix`, ...